hoomd/hpmc/shape_move.py

```diff
@@ -0,0 +1,256 @@
+
+    Args:
+        move_probability (`float`): Average fraction of shape parameters to
+            change each timestep.
+
+    Note:
+        Parameters must be given for every particle type and must be between 0
+        and 1. This class is limited to performing MC moves on the predefined
+        shape parameters, it does not performs any consistency checks internally.
+        Therefore, any shape constraint (e.g. constant volume, etc) must be
+        performed within the callback.
+
+    Example::
+
+        mc = hoomd.hpmc.integrate.ConvexPolyhedron()
+        mc.shape["A"] = dict(vertices=[(1, 1, 1), (-1, -1, 1),
+                                       (1, -1, -1), (-1, 1, -1)])
+        # example callback
+        class ExampleCallback:
+            def __init__(self):
+                default_dict = dict(sweep_radius=0, ignore_statistics=True)
+            def __call__(self, type, param_list):
+                # do something with params and define verts
+                return dict("vertices":verts, **self.default_dict))
+        python_move = hpmc.shape_move.PythonShapeMove()
+        python_move.callback = ExampleCallback()
+
+    Attributes:
+        callback (``callable`` [`str`, `list`], `dict` ]): The python function
+            that will be called to perform custom shape moves on arbitrary shape
+            parameters. The function must take the particle type and a list of
+            parameters as arguments and return a dictionary with the shape
+            definition whose keys **must** match the shape definition of the
+            integrator:
+
+                ``callable[[str, list], dict]``
+
+            Note that there is no type validation of the callback.
+
+        params (`TypeParameter` [``particle type``, `list`]): List of tunable
+            parameters to be updated.
+
+        move_probability (`float`): Average fraction of shape parameters to
+            change each timestep.
+    """
+
+    _suported_shapes = {
+        'ConvexPolyhedron', 'ConvexSpheropolyhedron', 'Ellipsoid'
+    }
+
+    def __init__(self, move_probability=1):
+
+        super().__init__(move_probability)
+        param_dict = ParameterDict(callback=object)
+        self._param_dict.update(param_dict)
+
+        typeparam_shapeparams = TypeParameter('params',
+                                              type_kind='particle_types',
+                                              param_dict=TypeParameterDict(
+                                                  [float], len_keys=1))
+        self._add_typeparam(typeparam_shapeparams)
+
+    @log(category='object', requires_run=True)
+    def type_params(self):
+        """dict: Tunable parameters as dictionary. The particle types are the
+                 keys of the dictionary.
+
+        Example:
+            >>> python_shape_move.type_params()
+            {'A': [0.21, 0.33], 'B': [0.561, 0.331, 0.123]}
+        """
+        return self._cpp_obj.getTypeParams()
```

Comment:
No need to have this I think.

Reply:
My intended purpose for this was to make the parameters loggable.

Reply:
This discussion came up on the MD alchemy pull request as well. We should not provide two names for the same thing. Remove the duplicate.

For now, users can log these with custom log quantities. If there is a strong need for these to be loggable without custom quantities, open an enhancement issue requesting that param_dict and TypeParameter quantities can be made loggable (optionally) and note which quantities should be added when the feature becomes available.